### ISSUE TITLE
[CAL-2150] + shouldn't be allowed in username, conflicts with dynamic bookings

### ISSUE THREAD

**Issue by hariombalhara** [calcom/cal.com#10078]
* "+ " shouldn't be allowed in username. It is reserved for dynamic bookings
  We should slugify + and replace it with '-'

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2150](https://linear.app/calcom/issue/CAL-2150/shouldnt-be-allowed-in-username-conflicts-with-dynamic-bookings)</sub>

**Comment by AyushGovil** [comment]
Hey @hariombalhara , can you elaborate the issue with a small example or share me the steps to reproduce?

**Comment by PeerRich** [comment]
we have dynamic. booking links such as cal.com/peer+bailey, thereby we need to ban "+" in usernames. adding a + should convert it to - in the username input

**Comment by PeerRich** [comment]
![CleanShot 2023-07-12 at 13 54 24@2x](https://github.com/calcom/cal.com/assets/8019099/f42e664f-fce2-4dac-9d9c-4c58b0f7343a)
luckily there are no users yet with "+" in their username

**Comment by AyushGovil** [comment]
I can work on this one if no one minds.

**Comment by AyushGovil** [comment]
Can someone check this => https://github.com/calcom/cal.com/pull/10142. Need review.

**Comment by valdemartti** [comment]
Although the related PRs have been closed, I believe this has been already implemented with `slugify` in the `updateProfileHandler` function and in the registration flow ([1eeb91a](https://github.com/calcom/cal.com/commit/1eeb91a793e577b10c3037a2158d7ad052659b20) and [f6005b8](https://github.com/calcom/cal.com/commit/f6005b8c70838421bf975aa876cbfde79b628bed)). I did notice that the username with `+` still persists in the user's session after changing a username, but believe that should be a different non-critical issue. Suggesting to close this issue as solved.
